internal/service/handlers/play.go

```diff
@@ -475,19 +475,62 @@ func handlePlayRequest(ctx context.Context, s *discordgo.Session, m *discordgo.M
 	urlStr := args["url"]
 
 	pid := p.PlaylistID()
+	playPack := make(chan service.PlayCall, 1)
 
-	if handled, err := processPlaylist(ctx, pid, s, m, p, urlStr); err != nil {
+	p.Enqueue(playPack)
+
+	var play func(as service.AudioStreamer)
+	{
+		mention := m.Author.Mention()
+		play = func(as service.AudioStreamer) {
+			playPack <- service.PlayCall{
```

Comment:
send should short circuit if the runtime root context is cancelled or if the async process context is canceled.